### PR TITLE
Ticketing router updates

### PR DESCRIPTION
**Pull request by arasmu2**
## Brief Description

Refactored to move SQL statements into dedicated functions in the .service file.
Removed unused routes.
Added date parsing functions to the db file.

## References Issue

Issue #

## Type of change

- [x] Bug fix
- [x] Feature
- [ ] Major change (potentially breaking, list anything that is broken below)
- [ ] Documentation

### Breaking Changes

None

## Testing done

Created the new route in Postman, checked that ticket type update works as expected. 

- [x] Docker was used.
- [ ] Docker was not used and I have listed my configuration below.

**System Specifications**:
* Operating System:
* Node Version:
* NPM Version:

## Checklist:

- [x] My code follows the styling guidelines.
- [ ] I have added unit tests and verified that they pass.
- [x] I have used the linter and fixed any linting issues.
- [x] I have commented the code.
- [x] I have adjusted the documentation to match the changed code.
- [x] Prior to submitting the PR, I made sure the latest changes were merged with my branch.
- [x] Existing tests pass locally with changes.

### Explanation for unchecked

Provide a brief explanation for why the checklist was not completed.

## Additional information: 

Provide any additional information that might be useful for this commit.

### CODE REVIEW

File: server/src/api/tickets/ticket.router.ts [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 
-export const ticketRouter = express.Router();
+export const ticketRouter = Router();
 
 
 //
 // **BROKEN** Will require refactor and DB update for
 //  eventinstance: defaulttickettype
 //
 // Responds with tickets subset of Redux state
```
Comment (ben16williams) [inline server/src/api/tickets/ticket.router.ts:15]: Can we remove this now?

File: server/src/api/tickets/ticket.router.ts [hunk 2]
```diff
@@ -28,13 +40,13 @@ ticketRouter.get('/', async (req, res) => {
...
                     ON et.eventinstanceid_fk = ei.eventinstanceid
-                  JOIN tickettype tt
+                  JOIN tickettype tt 
                     ON et.tickettypeid_fk = tt.tickettypeid
-                WHERE
+                WHERE 
                   ei.availableseats > 0
-                AND
+                AND 
                   ei.salestatus = true;`;
```
Comment (ben16williams) [inline server/src/api/tickets/ticket.router.ts:50]: Not a blocker but we should move this into a service function

Comment (ben16williams) [reply server/src/api/tickets/ticket.router.ts:50]: I'm actually not sure, can't we just use the router code that's commented out?

**Comment by arasmu2** [comment]
Moved the SQL to the service file and wrapped its results into the data filed of the JSON response to conform to best practices. I checked with Storm and he can make this work for the front end.

File: server/src/api/tickets/ticket.service.ts [hunk 3]
```diff
@@ -32,14 +80,68 @@ export const toTicket = (row:any): Ticket => {
...
+const reduceToTicketState = (res: any, t: Ticket) => {
   const id = t.event_instance_id;
   const {byId, allIds} = res;
   return allIds.includes(id) ?
     res :
     {byId: {...byId, [id]: t}, allIds: [...allIds, id]};
 };
 
+
+//
```
Comment (ben16williams) [inline server/src/api/tickets/ticket.service.ts:92]: Not a blocker but could add a short description, the function names are pretty descriptive though

**Review by ben16williams — APPROVED** [review]

**Pull request merged**